astroplan/constraints.py

```diff
@@ -394,7 +395,81 @@ def compute_constraint(self, times, observer, targets):
                              "MoonSeparationConstraint.")
         return mask
 
+    
+class DarkGreyBrightConstraint(Constraint):
+    """
+    Constrain whether it is dark, gray or bright time or how bright.
+    """
+    # TODO: make a new test
+    @u.quantity_input(horizon=u.deg)
+    def __init__(self, lunar_horizon=90*u.deg, min_moon_illumination=None,
+                                                max_moon_illumination=None):
+        '''
+        Parameters
+        ----------
+        Lunar_horizon : `~astropy.units.Quantity` (optional)
+            The altitude of the ``horizon`` and defaults to 90 degrees       
```

Comment:
The reason that I defaulted it to 90 degrees was the assumption that it would be used to define what times were considered dark, grey or bright. For dark time you care about the horizon, for grey and bright you care about moon illumination, but assume the moon is up. My intended functionality would actually be better served by including a minimum lunar altitude that is modifiable and overwritten if a maximum lunar altitude is defined.